integration/kubernetes/e2e_conformance/run.sh

```diff
@@ -66,48 +69,83 @@ run_sonobuoy() {
+	for entry in "${_skipped_tests[@]}"
+	do
+		_skip_options+=("${entry#- }|")
+	done
+
 	skip_options=$(IFS= ; echo "${_skip_options[*]}")
 	skip_options="${skip_options%|}"
 
-	sonobuoy run --e2e-skip="$skip_options" --wait="$WAIT_TIME"
-
-	# Retrieve results
-	e2e_result_dir="$(mktemp -d /tmp/kata_e2e_results.XXXXX)"
-	sonobuoy retrieve "$e2e_result_dir" || \
-		die "Couldn't retrieve sonobuoy results, please check status using: sonobuoy status"
-	pushd "$e2e_result_dir" >> /dev/null
 
-	# Uncompress results
-	ls | grep tar.gz | xargs tar -xvf
-
-	# e2e results log will be on "plugins/e2e" directory
-	e2e_result_log=$(find ./plugins/e2e -name "e2e.log")
-	info "Results of the e2e tests can be found on: $e2e_result_log"
-
-	# If on CI, display the e2e log on the console.
-	[ "$CI" == true ] && cat "$e2e_result_log"
+	if [ "${MINIMAL_K8S_E2E}" == "true" ]; then
+		FOCUS_TEST+="Secrets should be consumable via the environment"
+		FOCUS_TEST+="|ConfigMap should be consumable from pods in volume"
+		FOCUS_TEST+="|Projected secret should be consumable in multiple volumes"
+		FOCUS_TEST+="|Kubelet when scheduling a busybox command in a pod should printk"
+		FOCUS_TEST+="|InitContainer \[NodeConformance\] should invoke init containers"
+		sonobuoy run --e2e-focus="${FOCUS_TEST}" --e2e-skip="$skip_options" --wait="$WAIT_TIME"
+	else
+		sonobuoy run --e2e-skip="$skip_options" --wait="$WAIT_TIME"
+	fi
 
-	# Check for Success message on the logs.
-	grep -aq " 0 Failed" "$e2e_result_log" || die "A failure has been found"
-	grep -aq "SUCCESS" "$e2e_result_log" && \
-		info " k8s e2e conformance using Kata runtime finished successfully"
-	popd
+	e2e_result_dir="$(mktemp -d /tmp/kata_e2e_results.XXXXX)"
+	{
+		sonobuoy status --json
+		if ! results=$(sonobuoy retrieve "${e2e_result_dir}"); then
+			die "failed to retrieve results"
+		fi
+
+		sonobuoy results "${results}" --mode=dump
+
+		pushd "${e2e_result_dir}"
+		tar -xvf "${results}"
+		e2e_result_log=$(find "${e2e_result_dir}/plugins/e2e" -name "e2e.log")
+		info "Results of the e2e tests can be found on: $e2e_result_log"
+		popd
+
+		failed_query='.plugins | .[] | select( ."result-status" | contains("failed"))'
+		failed=$(sonobuoy status --json | jq "${failed_query}")
+		if [ "${failed}" != "" ]; then
+			if [ "$CI" == true ]; then
+				cat "$e2e_result_log"
+			fi
+			sonobuoy status --json | jq {failed_query}
+			die "Found failed tests in end-to-end k8s test"
+		fi
+		local jobs_file="${SCRIPT_PATH}/e2e_k8s_jobs.yaml"
+		local expected_passed_query="jobs.${CI_JOB:-}.passed"
+		local expected_passed=$("${GOPATH}/bin/yq" read "${jobs_file}" "${expected_passed_query}")
+		if [ "${expected_passed}" != "" ];then
+			passed_query='.plugins | [ .[]."result-counts".passed] | add'
+			passed=$(sonobuoy status --json | jq "${passed_query}")
+			if [ "${expected_passed}" != "${passed}" ];then
+				die "expected ${expected_passed} test are passing, but only ${passed} passed"
```

Comment:
rewording: `expected  ${expected_passed} tests to pass, but only  ${passed} passed`